packages/container/lib/container.js

```diff
@@ -447,16 +447,13 @@ function deprecatedFactoryFor(container, fullName, options = {}) {
     return factory;
   } else {
     let injections = injectionsFor(container, fullName);
-    let factoryInjections = factoryInjectionsFor(container, fullName);
-    let cacheable = !areInjectionsDynamic(injections) && !areInjectionsDynamic(factoryInjections);
-
-    factoryInjections[NAME_KEY] = registry.makeToString(factory, fullName);
+    let cacheable = !areInjectionsDynamic(injections);
 
     let injectedFactory = factory.extend(injections);
 
     // TODO - remove all `container` injections when Ember reaches v3.0.0
     injectDeprecatedContainer(injectedFactory.prototype, container);
-    injectedFactory.reopenClass(factoryInjections);
+    injectedFactory.reopenClass({ [NAME_KEY]: registry.makeToString(factory, fullName) });
```

Comment:
Ideally we can get rid of this also (I'm not sure we have solved the "problem" of factory injections if we cannot remove this `reopenClass`).